### ISSUE TITLE
Question: Working backward from existing Swagger spec?

### ISSUE THREAD

**Issue by bobspryn** [aws/chalice#906]
👋 

Building my first serverless application here and my head is swimming a bit. I have a defined swagger spec that my mobile client is working against. Swagger has the ability to generate code from a spec for flask, and other similar style frameworks in different languages.

Is there such a thing for chalice? Does that even make sense? Does having an existing spec net me anything here?

Should I use flask instead?

**Comment by sprynmr** [comment]
Right. So there is this: https://github.com/swagger-api/swagger-codegen

Which is used to take a spec and generate various bits for various clients/servers. The main python solution is using Connexion/Flask. It generates some model code, but mostly just hooks up Connexion which points requests in the right direction at runtime based on the swagger.yaml file.

Some solutions are dynamic like that, some just generate your starting stubs.

**Comment by bhrutledge** [comment]
[Connexion](https://connexion.readthedocs.io/en/stable/) is used by swagger-codegen, but stands on its own as a valdiation, routing, and serialization layer on top of Flask, driven by an existing OpenAPI spec as the source of truth, instead of generating one from Python code.

It looks like there's [work](https://github.com/spec-first/connexion/milestone/14) [underway](https://github.com/spec-first/connexion/issues/1395) to make the next major version of Connexion more pluggable, and independent of a particular framework. Perhaps pieces of that could be used in a Chalice app.
